elastic/kibana
# [Security solution] Second attempt at 8.8.1 GenAi Connector Feature Flag :)

## Summary

- Moves the registration call for the GenAi connector from `stack_connectors` to `security_solution` in order to apply the security solution `xpack.securitySolution.enableExperimental: ['assistantEnabled']` feature flag.
- Adds a technical preview label to the GenAi connector

### To test:

1. Checkout branch and run
2. Try to create a GenAI connector from the Stack Connectors area, note that it is not present in the connector list by default
3. Add the feature flag to dev yml `xpack.securitySolution.enableExperimental: ['assistantEnabled']`
4. Refresh and try to create a GenAI connector, you will now be able to do so

### Caveat:

If you create a GenAI connector and then disable the feature flag, it will show as a disabled connector. We decided to document the issue and leave this in as fixing it would require a second feature flag in `triggers_actions_ui`.
<img width="1728" alt="Screenshot 2023-06-01 at 8 39 54 PM" src="https://github.com/elastic/kibana/assets/6935300/1389c177-c95b-4251-b3c3-52a6797488e1">
